_posts/2022-01-03-logical-ops-on-java-collections.md

```diff
@@ -0,0 +1,344 @@
+In this article, we will look at some logical operations on Java Collections."
+image:
+  auto: 0074-stack
+---
+
+Collections are containers to group multiple items in a single unit. For example, a collection can represent a stack of books, products of a category, a queue of text messages, etc. They are an essential feature of almost all programming languages, most of which support different types of collections such as `List`, `Set`, `Queue`, `Stack`, etc. 
+
+Java also supports a rich set of collections packaged in the Java Collections Framework.
+
+In this article, we will look at examples of performing some useful operations on one or more Java Collections. 
+
+{% include github-project.html url="https://github.com/thombergs/code-examples/tree/master/core-java/collectionops" %}
+
+## Java Collections Framework
+A Collections Framework is a unified architecture for representing and manipulating collections. The Java Collections Framework is one of the core parts of the Java programming language. It provides a set of interfaces and classes to implement various data structures and algorithms along with several methods to perform various operations on collections. 
+
+The `Collection` interface is the root interface of the collections framework hierarchy.
+
+Java does not provide direct implementations of the `Collection` interface but provides implementations of its subinterfaces like `List`, `Set`, and `Queue`. While the official documentation of the [Java Collection Interface](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html) is the go-to guide for everything related to collections, here we will look at some examples of performing common logical operations like addition (joining), subtraction (difference), union, and the intersection between two or more collections. 
+
+## Adding one Collection to Another
+
+Let us start with the simplest of all: addition or joining two collections.
+![adding two collections](/assets/img/posts/logical-ops-java-coll/add.png)
+
+The [Stream](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html) class introduced since Java 8 provides useful methods for supporting sequential and parallel aggregate operations. In this example, we are performing the concatenation of elements from two collections using the `Stream` class:
+```java
+public class CollectionHelper {
+    
+    public List<Integer> add(final List<Integer> collA, 
+                             final List<Integer> collB){
+
+        return Stream.concat(
+                collA.stream(), 
+                collB.stream())
+            .collect(Collectors.toList());     
+    }   
+}
+```
+
+Here we are concatenating two collections in the `add` method in the `CollectionHelper`class. For adding, we have used the `concat()` method of the `Stream` class. We can also use this method to join more than 2 collections at a time. 
```

Comment:
When mentioning a method in the text, please add parentheses so you can see at first glance that it's a method and not a variable name. I changed it in most places already.